src/_document/render.tsx

```diff
@@ -43,7 +42,7 @@ function enhanceComponents(
   }
 
   return {
-    App: options.enhanceApp ? options.enhanceApp(App) : App,
```

Comment:
We should be able to get rid of this cast once we properly use `next/app` as the DefaultApp.